fl4health/clients/fenda_ditto_client.py

```diff
@@ -25,18 +26,23 @@ def __init__(
         device: torch.device,
         loss_meter_type: LossMeterType = LossMeterType.AVERAGE,
         checkpointer: Optional[ClientCheckpointModule] = None,
-        lam: float = 1.0,
+        metrics_reporter: Optional[MetricsReporter] = None,
+        progress_bar: bool = False,
         freeze_global_feature_extractor: bool = False,
     ) -> None:
         """
-        This client implements the Ditto algorithm from Ditto: Fair and Robust Federated Learning Through
-        Personalization with a local FENDA model. The idea is that we want to train a local FENDA model
-        along with the global model for each client. So we simultaneously train a global model that is
+        This client implements a combination of the Ditto algorithm from Ditto: Fair and Robust Federated Learning
+        Through Personalization with FENDA-FL models. The idea is that we want to train a local FENDA model
+        along with the global model for each client. We simultaneously train a global model that is
```

Comment:
In light of our conversation, perhaps adding this (no need to use words exactly): 
The global ditto model consists of a feature extractor and classification head, where the feature extractor shares the architecture of the global and local feature extractors for FENDA.
After this: The idea is that we want to train a local FENDA model along with the global model for each client.

And before getting into specifics about the nature of the constraint (because it depends on the mode of operation), be clear that there are two different modes of operation.
1. global DITTO model feature extractor SETS AND FREEZES weights of global FENDA feature extractor, local components of FENDA model are trained and an additional drift loss is computed between the local FENDA feature extractor and global feature extractor
2. global DITTO model feature extractor INITIALIZES weights of global FENDA feature extractor, both local and global components of FENDA are trained and a drift loss is calculated between DITTO global feature extractor and FENDA global feature extractor

Let me know if you think this is wrong or even more unclear!

Reply:
Nope that looks great. Added to the documentation.